compiler/packages/babel-plugin-react-compiler/src/CompilerError.ts

```diff
@@ -510,8 +532,25 @@ function printCodeFrame(
     },
     {
       message,
+      linesAbove: CODEFRAME_LINES_ABOVE,
+      linesBelow: CODEFRAME_LINES_BELOW,
     },
   );
+  const lines = printed.split(/\r?\n/);
+  if (loc.end.line - loc.start.line < CODEFRAME_MAX_LINES) {
+    return printed;
+  }
+  const pipeIndex = lines[0].indexOf('|');
+  return [
+    ...lines.slice(
+      0,
+      CODEFRAME_LINES_ABOVE + CODEFRAME_ABBREVIATED_SOURCE_LINES,
+    ),
+    '> ' + ' '.repeat(pipeIndex - 2) + '…',
```

Comment:
nit: should we add the pipe index here?

Currently we have
```
   2 | function Component(props) {
>  3 |   const data = useMemo(() => {
     |                        ^^^^^^^
>    …
>  7 |   }, [props.items?.edges?.nodes]);
     | ^^^^ Could not preserve existing manual memoization
```

Would it look better as this?
```
   2 | function Component(props) {
>  3 |   const data = useMemo(() => {
     |                        ^^^^^^^
>    | …
>  7 |   }, [props.items?.edges?.nodes]);
     | ^^^^ Could not preserve existing manual memoization
```